### ISSUE TITLE
Dynamic import of kolibri components fails 

### ISSUE THREAD

**Issue by fgeierst** [fgeierst/typo3-publicui-demo#1]
In Vite development mode, the Kolibri components mount, you can see the open shadow roots. However, the component specific entry point (kol-accordion.entry.js, kol-card.entry.js) can't be dynamically imported. Somehow Vite can't auto detect the dependency. 

-  I tried with [1.5.3](https://github.com/public-ui/kolibri/releases/tag/1.5.3) and [1.4.2](https://github.com/public-ui/kolibri/releases/tag/1.4.2), both showing the same behavior. 
- I found [`optimizeDeps.include`](https://vitejs.dev/guide/dep-pre-bundling.html#customizing-the-behavior), but that did not help me so far.
- To compare, I used the Kolibri CLI to look at a static site example. That loads the dependencies from https://esm.sh/@public-ui. But this too has 404 errors for component files like kol-accordion.entry.js.

@deleonio Do you have an idea how to resolve this?

<img width="581" alt="Screenshot 2023-08-12 at 11 27 16" src="https://github.com/fgeierst/typo3-publicui-demo/assets/22551442/3712ed02-2444-4a2a-87dd-15c977a7b431">
<img width="1147" alt="Screenshot 2023-08-12 at 11 28 15" src="https://github.com/fgeierst/typo3-publicui-demo/assets/22551442/5c2ef069-69c9-4788-a932-266f67f920b3">

**Comment by deleonio** [comment]
@fgeierst bitte mal `next`-Version ausprobieren.

`esm.sh` funktioniert seit geraumer Zeit nicht mehr.

**Comment by fgeierst** [comment]
@deleonio I updated to 1.6.0-rc.28, but still the same problem that component specific js files like kol-accordion.entry.js, kol-card.entry.js are not found. Is there a working demo that shows how public-ui is supposed to work outside of using React or Vue? I think esm.sh is not the cause, but how the component loader fetches the files and probably how to interact with the bunder (=Vite/Rollup).

**Comment by deleonio** [comment]
Hello @fgeierst,

please take a look at our `vite` examples. Use `npm init kolibri@next vite-react` and create a `vite-react` project.

You should integrate `KoliBri` like an asset - `index.html`:

```html
<!doctype html>
<html lang="de" dir="ltr">
	<head>
		<title>Vite React-App | KoliBri</title>
		<meta charset="UTF-8" />
		<link rel="icon" type="image/svg+xml" href="/favicon.ico" />
		<meta name="viewport" content="width=device-width, initial-scale=1.0" />
		<link rel="stylesheet" href="/assets/@public-ui/themes/assets/bundes/style.css" />
		<link rel="stylesheet" href="/assets/@public-ui/themes/assets/fontawesome-free/css/all.min.css" />
		<script type="module" src="/assets/@public-ui/components/dist/kolibri/kolibri.esm.js"></script>
	</head>
	<body>
		<div id="app"></div>
		<script type="module" src="/src/main.tsx"></script>
	</body>
</html>
```

The ESM import does not resolve the relative paths. But this is not a problem if you imagine that KoliBri registers web components in the DOM - we can do that outside your app code.

**Comment by fgeierst** [comment]
Thanks! I'll try that approach using React or Vue, too. 

For my original approach, I came one step closer: The dynamic import did not work, because Vite [excludes node_modules](https://github.com/vitejs/vite/issues/6179) by default. 
This can be solved by setting [`build.dynamicImportVarsOptions.exclude: []` ](https://github.com/fgeierst/typo3-publicui-demo/blob/c819d6e6cb6d95306da8dcd7886127d71e12a051/vite.config.js#L31).

Now the component files load, but there is a new issue: `TypeError: Cannot read properties of undefined (reading '$hostElement$')`
<img width="758" alt="Screenshot 2023-08-20 at 21 43 18" src="https://github.com/fgeierst/typo3-publicui-demo/assets/22551442/6d81b978-1ad6-459d-807c-4a0ddf73ed00">

**Comment by deleonio** [comment]
Good morning @fgeierst,

here a really tiny vite react kolibri app: https://github.com/deleonio/tiny-vite-react-kolibri-app

> Thanks for your vite config hint!

You need this line in your `index.html`:

```html
	<link rel="stylesheet" href="/node_modules/@public-ui/themes/assets/codicons/codicon.css" />
	<script type="module" src="/node_modules/@public-ui/components/dist/kolibri/kolibri.esm.js"></script>
</head>
```

And you does not need to define the custom elements not in the app context:

```tsx
import { register } from '@public-ui/components';
import { DEFAULT } from '@public-ui/themes';

import React from 'react';
import { createRoot } from 'react-dom/client';
import App from './App';

register(DEFAULT, []) // <- here a empty array
	.then(() => {
		const htmlElement: HTMLElement | null = document.querySelector<HTMLDivElement>('div#app');
		if (htmlElement instanceof HTMLElement) {
			const root = createRoot(htmlElement);
			root.render(
				<React.StrictMode>
					<App />
				</React.StrictMode>,
			);
		}
	})
	.catch(console.error);
```

**Comment by fgeierst** [comment]
Thanks for the react/vite demo app! I wonder if I need the React app at all or if I can't just use the web components without any UI framework. That would make more sense to me in a server rendered PHP based stack like TYPO3.

**Comment by fgeierst** [comment]
Resolved with 9d51d1205baee5369bbf133af38883e4c192bbed , Thanks @deleonio !
